src/test/java/net/lingala/zip4j/ExtractZipFileIT.java

```diff
@@ -280,6 +281,24 @@ public void testExtractFilesWithSetPasswordSetter() throws IOException {
     zipFile.extractAll(outputFolder.getCanonicalPath());
   }
 
+  @Test
+  public void testAddFolderWithNotNormalizedPath() throws IOException {
+    ZipFile zipFile = new ZipFile(generatedZipFile);
+    ZipParameters parameters = new ZipParameters();
+
+    try {
+      String folderToAddPath = TestUtils.getTestFileFromResources("").getPath() + InternalZipConstants.FILE_SEPARATOR + ".." + InternalZipConstants.FILE_SEPARATOR + TestUtils.getTestFileFromResources("").getName();
+      File folderToAdd = new File(folderToAddPath);
+      zipFile.addFolder(folderToAdd, parameters);
+      File fileToAdd = TestUtils.getTestFileFromResources("file_PDF_1MB.pdf");
+      String fileToAddPath = folderToAddPath + InternalZipConstants.FILE_SEPARATOR + fileToAdd.getName();
+      zipFile.addFile(fileToAddPath, parameters);
+      ZipFileVerifier.verifyZipFileByExtractingAllFiles(generatedZipFile, outputFolder, 13);
+    } catch (IOException e) {
+      fail("Should not throw an exception");
```

Comment:
This is not necessary. If there is an exception, this method will fail anyway. Please remove the catch block.

Reply:
Sure